tests/Feature/PhotosAddHandlerTestAbstract.php

```diff
@@ -344,4 +345,37 @@ public function testTrickyVideoUpload(): void
 			],
 		]);
 	}
+
+	/**
+	 * Tests video upload without ffmpeg or exiftool.
+	 *
+	 * @return void
+	 */
+	public function testVideoUploadWithoutFFmpeg(): void
+	{
+		$hasExifTool = Configs::getValueAsInt(self::CONFIG_HAS_EXIF_TOOL);
+		Configs::set(self::CONFIG_HAS_EXIF_TOOL, 0);
+
+		$hasFFMpeg = Configs::getValueAsInt(TestCase::CONFIG_HAS_FFMPEG);
+		Configs::set(TestCase::CONFIG_HAS_FFMPEG, 0);
+
+		$response = $this->photos_tests->upload(
+			TestCase::createUploadedFile(TestCase::SAMPLE_FILE_GAMING_VIDEO)
+		);
+		$response->assertJson([
+			'album_id' => null,
+			'title' => 'gaming',
+			'type' => TestCase::MIME_TYPE_VID_MP4,
+			'size_variants' => [
+				'original' => [
+					'width' => 0,
+					'height' => 0,
+					'filesize' => 66781184,
+				],
+			],
+		]);
+
+		Configs::set(self::CONFIG_HAS_FFMPEG, $hasFFMpeg);
+		Configs::set(self::CONFIG_HAS_EXIF_TOOL, $hasExifTool);
+	}
```

Comment:
Maybe we should additionally assert that the expected error message, e.g. "FFMpeg is disabled by configuration", can be found in the logs?

To this end we could clear the log table at the beginning of the test with `DB::tables('logs)->delete()`. At the end of the test we do a count-query and assert that we found an entry, e.g. something like

```php
$numErrorMsg = DB::tables('logs)->where('message', 'like', '%FFMpeg is disabled by configuration%')->count();
self::assertEquals(1, $numErrorMsg);
```

But maybe this would make the test too strict? I am unsure.

Reply:
I wouldn't do it myself (testing for log messages seems a little overly prescriptive to me) but feel free to add a commit to that effect to this PR if you want.